etcdmain/etcd.go

```diff
@@ -53,7 +62,17 @@ func Main() {
 
 	var stopped <-chan struct{}
 
-	shouldProxy := cfg.isProxy()
+	if cfg.dir == "" {
+		cfg.dir = fmt.Sprintf("%v.etcd", cfg.name)
+		log.Printf("etcd: no data-dir provided, using default data-dir ./%s", cfg.dir)
+	}
+
+	which := identifyDataDirOrDie(cfg.dir)
+	if which != dirEmpty {
+		log.Printf("etcd: %v dir already exists, starting etcd %v...", which, which)
```

Comment:
x dir is a type not a path. `member dir` exists -> member; `proxy dir` exists -> proxy.